ADAPT-2397: Mobile main nav

yvonnetangsu opened the pull request:
# READY FOR REVIEW

# Summary
- Mobile main nav 
- Supports spacebar/enter to open/close, escape key to close, tabbing, and click outside main nav to close.
- For this PR, please review on Chrome, as it's buggy on Firefox and Safari so I'll submit another PR to swap out HeadlessUI.

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test
Figma mock:
![Screen Shot 2021-07-12 at 9 04 39 AM](https://user-images.githubusercontent.com/42749717/125320175-49049e80-e2f0-11eb-97b5-3c4b928ec525.png)


1. https://deploy-preview-69--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-hero/
2. Check the mobile nav on XS-MD breakpoints. It should function in the same way as the Giving site mobile menu.
2. For this PR, please review on Chrome, as it's buggy on Firefox and Safari so I'll submit another PR to swap out HeadlessUI.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

yvonnetangsu commented:
https://share.getcloudapp.com/NQuoBAbJ

@sherakama Looks like Firefox and Safari don't like nested popover 😬  - see above video. So I'll leave the parent item/drop down as is, but for the hamburger hide/show functionality, I'm going to use the Giving way of React hooks instead of using HeadlessUI since that seems more reliable for now.

yvonnetangsu commented:
@sherakama I seem to be having some issue with my local build even though the netlify preview seems ok. I've tried a lot of things (nuke, clean cache, reinstall, update plugins - gatsby postcss, change TW version, gatsby etc, but that didn't work either. I re-download the whole repo and build from scratch and that worked the first time I build, but if I attempt any changes to TW config and do npm run build again, then I got this weird result again 😢 
Here's what my local looks like
![Screen Shot 2021-07-08 at 10 11 44 PM](https://user-images.githubusercontent.com/42749717/125026594-c9e64080-e039-11eb-9587-d32a4f3a9ef4.png)

And here are the errors
![Screen Shot 2021-07-08 at 10 12 31 PM](https://user-images.githubusercontent.com/42749717/125026601-ceaaf480-e039-11eb-9d2c-92fbdc18ae8d.png)

sherakama commented:
I got that on this repo once before as well. I never did figure out the cause of it but I had to do a full nuke and destroy the whole directory and re-clone it all back to get it going again. All that my research came up with was an issue somewhere with postcss and autoprefixer that an upgrade to a version was supposed to fix. Try the full nuke. Don't forget to save your .env files first.

yvonnetangsu commented:
> I got that on this repo once before as well. I never did figure out the cause of it but I had to do a full nuke and destroy the whole directory and re-clone it all back to get it going again. All that my research came up with was an issue somewhere with postcss and autoprefixer that an upgrade to a version was supposed to fix. Try the full nuke. Don't forget to save your .env files first.

Yikes. Yeah, I did a fresh clone, and that worked the first time I build, but got this again in subsequent build (eg, when I change any TW utility) 😬  I guess I shall do it one more time and cross finger that I won't be needing any new utilities (should be very few now)

sherakama commented:
If it comes up again, post an issue to the TW github page and we can spend a little more time troubleshooting it.

yvonnetangsu commented:
> If it comes up again, post an issue to the TW github page and we can spend a little more time troubleshooting it.

Thanks, Shea! Will do.

sherakama commented:
This is looking good. Quick first keyboard test and I can tab focus on the hidden nav items when the menu is collapsed. I should not be able to do this.

yvonnetangsu commented:
> This is looking good. Quick first keyboard test and I can tab focus on the hidden nav items when the menu is collapsed. I should not be able to do this.

Good catch! Let me make sure they're hidden when menu is collapsed

sherakama commented on src/components/navigation/navItem.js:
```diff
@@ -1,10 +1,16 @@
 
-const NavItem = ({ blok: { classes, link, text }, blok, hasExternalIcon }) => (
+const NavItem = ({
+  blok: { classes, link, text },
```
What is this `classes`? We don't support css classes directly from storyblok now do we? Looks like the options tab is empty in storyblok. What where you doing with this? 
![Screen Shot 2021-07-13 at 10 06 35 AM](https://user-images.githubusercontent.com/550602/125495524-3bff30d9-17d9-4274-84dd-581c6a156215.png)

yvonnetangsu replied:
Yeah I think I can remove it. It was added very early on.

sherakama commented on src/components/navigation/mainNav.js:
```diff
@@ -1,26 +1,79 @@
+      </nav>
+      <nav
+        className="main-nav-mobile lg:su-hidden"
+        aria-label="Main Menu"
```
Is it ok to who two nav blocks side by side with the same aria label?

yvonnetangsu replied:
It should be ok since the `su-hidden` class (`display: none`) will hide things completely from the DOM so screen readers would not be able to "see" both navs at the same time.

sherakama reviewed (approved):
Please review the comments and revise the tab index issues and then this is GTG.

yvonnetangsu commented:
> This is looking good. Quick first keyboard test and I can tab focus on the hidden nav items when the menu is collapsed. I should not be able to do this.

The issue is now fixed. I removed the "su-hidden" that I initially added to the collapsed menu, since in order to have animation, one can't have "su-hidden/display:none" or you won't see any animation (the expanding). I have added `su-invisible` class when the menu is collapsed, which still hides it completely from DOM but allows for the expanding animation.
